staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3763,8 +3851,13 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 	}
 
 	if isNLB(annotations) {
-
-		if path, healthCheckNodePort := servicehelpers.GetServiceHealthCheckPathPort(apiService); path != "" {
+		customHealthCheck := func(annotations map[string]string) bool {
+			_, customProtocol := annotations[ServiceAnnotationLoadBalancerHealthCheckProtocol]
+			_, customPort := annotations[ServiceAnnotationLoadBalancerHealthCheckPort]
+			_, customPath := annotations[ServiceAnnotationLoadBalancerHealthCheckPath]
+			return customProtocol || customPort || customPath
+		}
+		if path, healthCheckNodePort := servicehelpers.GetServiceHealthCheckPathPort(apiService); path != "" && !customHealthCheck(annotations) {
 			for i := range v2Mappings {
```

Comment:
btw, i know the code is originally structured like this. but can we move this special handling for local traffic mode into parseNLBHealthCheckAnnotations directly?(and rename it like `buildNLBHealthCheckSetting`)
That way it's easier to unit test the behavior.

Reply:
sounds good